File: Python/selectionSort.py
"""
one of the easiest sorting algorithm

1. at first we need to find out the smallest value
2. replace it with the 1st unsorted element
3. increase index after exchange values positions.

"""

def selectionSort(list):
    minIndex = 0
    for i in range(len(list)-1):
        minIndex=i
        for j in range(i+1, len(list)):
            if(list[minIndex]>list[j]):
                minIndex = j
        if(minIndex!=i):
            temp = list[i]
            list[i]=list[minIndex]
            list[minIndex]=temp
    return list

list1 = list(map(int, input("Enter all elements of the list: ").split()))

print(f"Initial list: {list1}")
print(f"Sorted list: {selectionSort(list1)}")

File: Dart/selectionSort.dart
/*
one of the easiest sorting algorithm

1. at first we need to find out the smallest value
2. replace it with the 1st unsorted element
3. increase index after exchange values positions.

*/

import 'Dart:io';

List<int> selectionSort({
  required List<int> list,
  required int index,
}) {
  int minIndex = 0;
  int temp = 0;
  for (int i = 0; i < index - 1; i++) {
    minIndex = i;
    for (int j = i + 1; j < index; j++) {
      if (list[j] < list[minIndex]) {
        minIndex = j;
      }
    }
    if (i != minIndex) {
      temp = list[i];
      list[i] = list[minIndex];
      list[minIndex] = temp;
    }
  }
  return list;
}

void main() {
  print("Enter index number of a list: ");
  int? indexNo = int.parse(stdin.readLineSync()!);

  List<int> list = [];
  print("Enter elements of the list:");
  for (int i = 0; i < indexNo; i++) {
    list.add(int.parse(stdin.readLineSync()!));
  }
  print("Initial list: $list");
  List<int> sortedList = selectionSort(list: list, index: indexNo);
  print("Sorted list: $sortedList");
}
